### PR TITLE
Proper use partialDataset instead of runRange=0 #7135

### PR DESCRIPTION
**Pull request by novicecpp**
Solve https://github.com/dmwm/CRABServer/issues/7135

Stefano already adds the ability to submit a job with data partially on disk in https://github.com/dmwm/CRABServer/commit/6ae0db121c229838a2782ce2e00c4b224de735f7, but in the hack way by set `Data.runRange` to 0.

This PR is proper accept new configuration argument `partialdataset` (`Data.partialDataset` in client https://github.com/dmwm/CRABClient/pull/5168).

As Stefano suggested in the issue, instead of adding `partialdataset` as a new column in the database, I added a generic CLOB column called `tm_user_config` to store user config in key/value manner and named the new config to `partialdataset` to match the variable name API received. We can migrate another config to this column later.

Note that the downside of putting it into KV is we cannot filter in the database anymore (for instance, you cannot write SQL to select the only task that has `partialdataset` set to `True`), so we need to be careful put/migrate config to this KV.

### CODE REVIEW

**Comment by novicecpp** [comment]
@belforte Please review.

**Comment by belforte** [comment]
ASAP ! sorry

**Comment by belforte** [comment]
thanks for clarifications. Yes, we need to add one test to ClientValidation. But it may be easier to do it after it is deployed in production so that tests do no break if using prod client. That test is aimed to keep code stable, not to validate new features.

Support for `{k:v}` list in configuration is not needed now, I agree. It is more for things like https://github.com/dmwm/CRABServer/issues/6534

Indeed adding the new column by "hand" from SQLPLUS is the only way. We need to do it in all DB instances.

Bottom line: put the hack back (there are a couple users who reported using it, let's not confuse them), update all DB instance, merge, deploy in preprod and run all Jenkins tests simply to exercise as many REST API's as we can. I do not expect any problem, changes look good, but testing is always better.

**Comment by novicecpp** [comment]
Ok I will do it.
I need to discuss how we manipulate with db with you later though.

**Comment by belforte** [comment]
thanks for the backport. Looks good. We can talk about DB whenever you feel like

**Review by belforte — APPROVED** [review]

**Comment by belforte** [comment]
Looks good you may merge. But we need to recap deployment plan. Maybe here ? Maybe in the original issue ?

IIUC this is compatible with current client, so we do not worry about that. But steps should be:
1. add column to DB
2. deploy new REST
3. deploy new TaskWorker

correct ?

**Comment by novicecpp** [comment]
Yes. 
I will add column to prod db next Monday morning and deploy it if there is no problem.
